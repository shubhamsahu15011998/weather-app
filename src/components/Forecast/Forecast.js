import React, { useState } from 'react';

const Forecast = () => {

    let [responseObj, setResponseObj] = useState({});

    function getForecast() {
        fetch("https://community-open-weather-map.p.rapidapi.com/weather?q=Seattle", {
                "method": "GET",
                "headers": {
                    "x-rapidapi-key": "115e5d084dmshc7574d0783db844p1da38djsne98e5d501609",
                    "x-rapidapi-host": "community-open-weather-map.p.rapidapi.com"
                }
            })
            .then(response => response.json())
            .then(responseObj => {
                setResponseObj(responseObj)
            })
            .catch(err => {
                console.error(err);
            });
    }

    return ( <
        div >
        <
        h2 > Find Current Weather Conditions < /h2> <
        div > { JSON.stringify(responseObj) } <
        /div> <
        button onClick = { getForecast } > Get Forecast < /button> <
        /div>
    )
}

export default Forecast;