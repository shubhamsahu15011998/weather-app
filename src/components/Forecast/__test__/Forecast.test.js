import React from 'react';
import {
    cleanup,
    render,
    screen
} from '@testing-library/react';
import ReactDOM from 'react-dom';
import Forecast from "../Forecast"
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("Sapshot Matching", () => {

            const tree = renderer.create( < Forecast > < /Forecast>).toJSON();
                expect(tree).toMatchSnapshot();
            });

        test('Renders Input Label correctly', () => {
                render( < Forecast > < /Forecast>);
                    expect(screen.getByPlaceholderText("Enter City")).toBeInTheDocument();
                });

            test('Renders Radio Button 1 Label correctly', () => {
                    render( < Forecast > < /Forecast>);
                        expect(screen.getByText("Fahrenheit")).toBeInTheDocument();
                    });

                test('Renders Radio Button 2 Label correctly', () => {
                        render( < Forecast > < /Forecast>);
                            expect(screen.getByText("Celcius")).toBeInTheDocument();
                        });

                    test('Renders Submit Button correctly', () => {
                            render( < Forecast > < /Forecast>);
                                expect(screen.getByText("Get Forecast")).toBeInTheDocument();
                            });