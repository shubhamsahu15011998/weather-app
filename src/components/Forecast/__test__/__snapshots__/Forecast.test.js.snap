// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sapshot Matching 1`] = `
<div>
  <h2>
     Find Current Weather Conditions 
  </h2>
   
  <form
    onSubmit={[Function]}
  >
    <input
      className="textInput"
      maxLength="50"
      onChange={[Function]}
      placeholder="Enter City"
      type="text"
      value=""
    />
     
    <label
      className="Radio"
    >
      <input
        checked={true}
        name="units"
        onChange={[Function]}
        type="radio"
        value="imperial"
      />
      Fahrenheit 
    </label>
     
    <label
      className="Radio"
    >
      <input
        checked={false}
        name="units"
        onChange={[Function]}
        type="radio"
        value="metric"
      />
      Celcius 
    </label>
    <button
      className="Button"
      type="submit"
    >
      Get Forecast 
    </button>
     
  </form>
   
  <div>
    <div>
      <p>
         
        <strong>
           
           
        </strong>
      </p>
    </div>
     
  </div>
   
</div>
`;
